mailpoet/assets/js/src/automation/listing/index.tsx

```diff
@@ -21,56 +93,64 @@ export function AutomationListing({ workflows, loading }: Props): JSX.Element {
   ];
 
   return (
-    <TableCard
-      className="mailpoet-automation-listing"
-      title=""
-      isLoading={workflows.length === 0 || loading}
-      headers={headers}
-      rows={rows}
-      rowKey={(_, i) => workflows[i].id}
-      query={{ page: 2 }}
-      rowsPerPage={7}
-      totalRows={workflows.length}
-      hasSearch
-      showMenu={false}
-      actions={[
-        <div key="actions" className="mailpoet-automation-listing-actions">
-          <Dropdown
-            renderToggle={({ isOpen, onToggle }) => (
-              <Button
-                className="mailpoet-automation-listing-action-button"
-                variant="secondary"
-                aria-expanded={isOpen}
-                onClick={onToggle}
-                label=""
-                disabled
-              >
-                Actions
-              </Button>
-            )}
-            renderContent={() => (
-              <MenuGroup>
-                <MenuItem icon="trash">Move to trash</MenuItem>
-              </MenuGroup>
-            )}
+    <TabPanel
+      className="mailpoet-filter-tab-panel"
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore - the Tab type actually expects a string for titles but won't render HTML,
+      // making it very difficult to style the count badges. It seems to be compatible wtih JSX
+      // elements, however.
+      tabs={tabs}
```

Comment:
This might be one of the biggest issues. I'm not sure how we would get the styled badge counts if we have to use a normal string for the tab titles

Reply:
This [looks safe](https://github.com/WordPress/gutenberg/blob/39172d1aa22adfa0b337de6bb53801276f868065/packages/components/src/tab-panel/index.js#L87), at least for now. Maybe [the DefinitelyTyped types](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/fb66211010980641f01bcfe8ffbc0cc3d71989ca/types/wordpress__components/tab-panel/index.d.ts#L12) are even wrong.

There are also some [other fields](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/fb66211010980641f01bcfe8ffbc0cc3d71989ca/types/wordpress__components/tab-panel/index.d.ts#L21) but I guess those help only in the tab body and not in the title?

Reply:
This might be solvable also via CSS, using some pseudoelement like `::after` and passing in value from some data attribute, maybe. But it's weird and probably not very accessible.

Reply:
I would prefer to avoid a CSS solution like that if possible, but I also don't want this to cause any issues, or really require the `@ts-ignore` at all.

> There are also some other fields but I guess those help only in the tab body and not in the title?

This matches my understanding. You can attach arbitrary key/value pairs but they're only intended to be used in the callback that renders the tab body.

I've refactored to an approach that doesn't require using JSX for the `title` (even though it's probably totally fine), but I doubt that it's doing things the "react" way. I would appreciate your guidance on how to make it better if you have any ideas!

Reply:
Very good points! I've converted back to the JSX for now.